### PR TITLE
Add rule : color-no-named

### PR DESCRIPTION
**Pull request by makotot**
ref : #424
> is-named-css-color might be useful here.

I used [is-css-color-name](https://www.npmjs.com/package/is-css-color-name).
[is-named-css-color](https://www.npmjs.com/package/is-named-css-color) returns `true` if the color value such as `$color-name-black` and `var(--color-name-white)`. I think that the variables like `$color-name-black` should be valid in this rule because the variable that contains with color name is not always color name.

### CODE REVIEW

File: src/rules/color-no-named/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const testRule = ruleTester(rule, ruleName)
+
+testRule(undefined, tr => {
+  warningFreeBasics(tr)
+
+  tr.ok("a { color: #000; }")
+  tr.ok("a { color: #ababab; }")
+  tr.ok("a { color: rgba(0, 0, 0, 0); }")
+  tr.ok("a { something: #000, #fff, #333; }")
+
```
Comment (davidtheclark) [inline src/rules/color-no-named/__tests__/index.js]: Please add tests ensuring that color names are ignored if they are within comments, strings, or urls without quotes.

File: src/rules/color-no-named/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  tr.ok("a { padding: 000; }")
+  tr.ok("a::before { content: \"#ababa\"; }")
+
+  tr.ok("a { border-#$side: 0; }", "ignore sass-like interpolation")
+  tr.ok("a { box-sizing: #$type-box; }", "ignore sass-like interpolation")
```
Comment (davidtheclark) [inline src/rules/color-no-named/__tests__/index.js]: I do not understand the purpose of these two tests -- can you explain?

File: src/rules/color-no-named/__tests__/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  tr.ok("a { color: $black; }", "ignore sass-like interpolation")
```
Comment (davidtheclark) [inline src/rules/color-no-named/__tests__/index.js]: Not really `interpolation`, right? Just variable name?

File: src/rules/color-no-named/__tests__/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  tr.ok("a { color: var(--some-color-blue); }", "ignore css variable named with color")
+
```
Comment (davidtheclark) [inline src/rules/color-no-named/__tests__/index.js]: Please also test keyframe animation names that include colors, e.g. `animation: spin-blue 2s linear;`

**Comment by davidtheclark** [comment]
Awesome @makotot : it's great to have more hands on deck! I just asked for a few more tests.

File: src/rules/color-no-named/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+export default function (actual) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { value } = decl
+
+      valueParser(value).walk(function (node) {
+        const charsBeforeColon = decl.toString().indexOf(":")
```
Comment (jeddy3) [inline src/rules/color-no-named/index.js]: Shall we pull this out into a utility? As it's going to get used a lot.

Perhaps something along the lines of `declarationValueIndexOffset` to match the existing [`mediaQueryParamIndexOffset`](https://github.com/stylelint/stylelint/blob/master/src/utils/mediaQueryParamIndexOffset.js)?

**Comment by jeddy3** [comment]
@makotot Like @davidtheclark said, thanks for contributing! I've made one suggestion above.

File: src/rules/color-no-named/index.js (hunk 5 above)
Comment (davidtheclark) [inline src/rules/color-no-named/index.js]: Sounds good to me.

File: src/rules/color-no-named/__tests__/index.js (hunk 2 above)
Comment (makotot) [inline src/rules/color-no-named/__tests__/index.js]: Sorry. It is my mistake.

File: src/rules/color-no-named/__tests__/index.js (hunk 3 above)
Comment (makotot) [inline src/rules/color-no-named/__tests__/index.js]: Sorry. I misunderstood the meanings of `interpolation`. As you say, just variable name.

**Comment by makotot** [comment]
@davidtheclark @jeddy3 
Thanks :) I have updated PR.

**Comment by jeddy3** [comment]
@makotot Thanks for making the changes! Looks great to me.

@davidtheclark Not sure what's going on with that lint error on travis and not appveyor. If you're happy with the changes, shall I merge it anyway in the hope the error goes away? If it doesn't I'll look into after the merge.

**Comment by davidtheclark** [comment]
@jeddy3 : Sounds good.

**Comment by jeddy3** [comment]
Merged :tada: 

Thanks @makotot  !

**Pull request merged**